tools/build_chain.sh

```diff
@@ -1523,13 +1527,13 @@ groups=
 groups_count=
 for line in ${ip_array[*]};do
     ip=${line%:*}
-    num=${line#*:}
-    if [ -z $(echo $ip | grep -E "^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$") ];then
+    num=${line##*:}
+    if [[ ! $(echo $ip | grep -qE "^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$") && -z "${use_ipv6}" ]];then
         LOG_WARN "Please check IP address: ${ip}, if you use domain name please ignore this."
     fi
     [ "$num" == "$ip" ] || [ -z "${num}" ] && num=${node_num}
-    echo "Processing IP:${ip} Total:${num} Agency:${agency_array[${server_count}]} Groups:${group_array[server_count]}"
-    [ -z "$(get_value ${ip//./}_count)" ] && set_value ${ip//./}_count 0
+    echo "Processing IP=${ip} Total=${num} Agency=${agency_array[${server_count}]} Groups=${group_array[server_count]}"
+    [ -z "$(get_value ${ip//[\.:]/_}_count)" ] && set_value "${ip//[\.:]/_}_count" 0
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5837713)